OSDOCS-18128: adds intro and arch to MCP gateway (#109317)
openshift/openshift-docs: modules/con-about-mcp-gateway.adoc

// Module included in the following assemblies:
//
// *mcp_gateway_discover/about-mcp-gateway.adoc

:_mod-docs-content-type: CONCEPT
[id="con-about-mcp-gateway_{context}"]
= About the {mcpg}

[role="_abstract"]
You can focus on your agentic AI systems without building networking into your application code by using the Model Context Protocol (MCP) gateway. Application teams and platform engineers can operate and collaborate to expose MCP servers with the MCP gateway as secure and protected services just as you do with existing RESTFul based APIs.

For example, you can achieve the following by using the MCP gateway to connect your MCP servers:

* Aggregate MCP servers behind a single endpoint.
* Grow your agentic AI applications at scale, keeping connectivity outside of application development.
* Manage access to and the security of AI tools and MCP servers.

The {prodname} implementation of the MCP gateway extends the benefits of the Envoy proxy server to your customized AI agent systems.

> The Envoy proxy server intercepts all network traffic between your backend MCP servers and agentic AI systems. Envoy is a conformance-tested implementation of the Kubernetes Gateway API. By using the MCP gateway, you can extend the capabilities of Envoy without the complexity of manual configuration.

Reviewer: might be just me but I think this might overstate the scope of the gateway.

The gateway handles ingress from clients to MCP servers, not all traffic between services. perhaps something like (feel free to reword):

```suggestion
The Envoy proxy server handles traffic from agentic AI clients to your backend MCP servers at the gateway ingress. Envoy is a conformance-tested implementation of the Kubernetes Gateway API. By using  the MCP gateway, you can extend the capabilities of Envoy without the complexity of manual configuration.
```
